src/Microsoft.AspNetCore.Antiforgery/DefaultAntiforgeryTokenGenerator.cs

```diff
@@ -40,7 +41,17 @@ public AntiforgeryToken GenerateCookieToken()
                 throw new ArgumentNullException(nameof(httpContext));
             }
 
-            Debug.Assert(IsCookieTokenValid(cookieToken));
+            if (cookieToken == null)
+            {
+                throw new ArgumentNullException(nameof(cookieToken));
+            }
+
+            if (!IsCookieTokenValid(httpContext, cookieToken))
+            {
+                throw new ArgumentException(
+                    Resources.Antiforgery_CookieToken_MustBeProvided_Generic,
```

Comment:
I'll just add a new message since none of these really seem like an exact fit.

`IsValidCookieToken` is extensible so, it's probably for the best.